mono/tests/bug-gh-9507.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+
+public struct TestIsInst<T>
+{
+    public T[] inArray1;
+
+    public TestIsInst(int i) {
+        inArray1 = new T[16];
+
+		if (inArray1 is double[])
+			Console.WriteLine ("OK.");
```

Comment:
this test is a little mysterious.  I guess the `is double[]` is what's important here?  the Console Write is misleading then.  maybe `throw new Exception ("Unexpected")` ?

Also maybe a comment with a clue of how this relates to `MonoClass:cast_class`?

Reply:
Yeah I copy pasted the test from the issue, I'll make it nicer. I'm writing a nicer commit message too. I'll resolve when they're ready